main_test.go

```diff
@@ -89,8 +95,8 @@ func TestHandlePostMissingParams(t *testing.T) {
 			t.Errorf("failed to decode response body %v", err)
 		}
 
-		if resp.ErrCode != "M_BAD_JSON" {
-			t.Errorf("unexpected error code: got %v want %v", resp.ErrCode, "M_BAD_JSON")
+		if resp.ErrCode != "M_INVALID_PARAM" {
```

Comment:
Nitpick but:

`M_BAD_JSON` is:

>These error codes can be returned by any API endpoint:
>M_BAD_JSON Request contained valid JSON, but it was malformed in some way, e.g. missing required keys, invalid values for keys.

`M_INVALID_PARAM`:
>The following error codes are specific to certain endpoints.
>M_INVALID_PARAM A parameter that was specified has the wrong value. For example, the server expected an integer and instead received a string.

Arguably `M_BAD_JSON` is a better fit since we're mostly checking for missing keys in our validation layer, and sometimes the value.   `M_INVALID_PARAM` says in the spec it's for specific endpoints.